File: day 8/phonebook.cs
using System;
using System.Collections.Generic;
using System.IO;
class Solution {
    static void Main(String[] args) {
        int n;
        n = Int32.Parse(Console.ReadLine());

        Dictionary<string, string> phoneBook = new Dictionary<string, string>();
        for (int i = 0; i < n; i++){
            string name_number = Console.ReadLine();

            string[] seperated = name_number.Split(' ');

            phoneBook.Add(seperated[0], seperated[1]);
        }

        List<string> names = new List<string>();
        string line;
        while ((line = Console.ReadLine()) != null) {
            names.Add(line);
        }   

        foreach (string name in names) // Loop through List with foreach
        {
            if (phoneBook.ContainsKey(name))
                Console.WriteLine(name+"="+phoneBook[name]);
            else
                Console.WriteLine("Not found");

        }
    }
}

File: day 8/phonebook.py
"""Create and query a rudimentary phonebook."""
import sys

def phonebook():
    """Return a phonebook.

    First take an integer representing
    the amount of names and numbers to be added.
    Then take that amount of lines with space-separated
    strings.
    """
    number = int(input())
    phone_book = dict()
    for _ in range(number):
        name, number = input().split()
        phone_book[name] = number
    return phone_book

def get_names():
    """Take in an unknown amount of names."""
    names = list()
    for line in sys.stdin:
        names.append(line.rstrip())
    return names

def query_phonebook(phone_book, names):
    """Query the phonebook with a list of names."""
    for name in names:
        if name in phone_book:
            print(name + "=" + phone_book[name])
        else:
            print("Not found")

if __name__ == "__main__":
    PHONE_BOOK = phonebook()
    NAMES = get_names()
    query_phonebook(PHONE_BOOK, NAMES)
